# FASTA Analyzer

A Python script to analyze DNA sequences from FASTA files.  
It validates sequences, calculates their length, base composition (A, T, G, C), and GC content percentage.  
Results are saved into a CSV file for easy viewing and downstream analysis.

---

## Features
- Reads multi‚Äësequence FASTA files
- Validates sequences using regex (`^[ATGC]+$`)
- Calculates:
  - Sequence length
  - Base composition (A, T, G, C counts)
  - GC content percentage
- Outputs results into a clean CSV file

---

## Project Structure
FASTA-Analyzer/ 
‚îÇ 
‚îú‚îÄ‚îÄ sequence_multi.fasta   # Example FASTA input file
‚îú‚îÄ‚îÄ fasta_analyzer.py      # Main Python script 
‚îú‚îÄ‚îÄ sequence_multi.csv     # Output CSV file 
|‚îÄ‚îÄ README.md              # Project documentation


---

## üõ†Ô∏è Requirements
- Python 3.x
- No external libraries required (uses built‚Äëin `re` and `csv` modules)

---

## Usage
1. Clone the repository:
   bash
   git clone https://github.com/pijushrkmvm-spec/FASTA-Analyzer.git
   cd FASTA-Analyzer
   
2. Run the script
   python fasta_analyzer.py

---

## Learning GoalsThis project was part of my Python learning pathway:
 
- Practicing file handling
- Using regex for validation
- Working with lists and CSV output
- Applying bioinformatics concepts (GC content)

---

## Future Improvements- Add motif search (e.g., ATG start codon, TATA box) (Upcoming...)
- Support protein FASTA files
- Build a simple command‚Äëline interface (CLI)

---

## AuthorPijush Chakraborty
M.Sc. Bioinformatics student | Python learner | Interested in bioinformatics tools and automation
